tools/fork-network/src/cli.rs

```diff
@@ -0,0 +1,253 @@
+pub struct ForkNetworkCommand {
+    #[arg(short, long)]
+    pub reset: bool,
+    #[arg(short, long, default_value = "1000")]
+    pub epoch_length: u64,
+}
+
+impl ForkNetworkCommand {
+    pub fn run(
+        self,
+        home_dir: &Path,
+        genesis_validation: GenesisValidationMode,
+    ) -> anyhow::Result<()> {
+        if self.reset {
+            let near_config = load_config(home_dir, GenesisValidationMode::UnsafeFast)
+                .unwrap_or_else(|e| panic!("Error loading config: {:#}", e));
+
+            let store_path = home_dir
+                .join(near_config.config.store.path.clone().unwrap_or(PathBuf::from("data")));
+            let fork_snapshot_path = store_path.join("fork-snapshot");
+            if !Path::new(&fork_snapshot_path).exists() {
+                panic!("Fork snapshot does not exist");
+            }
+            println!("Removing all current data");
+            for entry in std::fs::read_dir(&store_path)? {
+                let entry = entry?;
+                if entry.file_type()?.is_file() {
+                    std::fs::remove_file(&entry.path())?;
+                }
+            }
+            println!("Restoring fork snapshot");
+            for entry in std::fs::read_dir(&fork_snapshot_path)? {
+                let entry = entry?;
+                std::fs::rename(&entry.path(), &store_path.join(entry.file_name()))?;
+            }
+            std::fs::remove_dir(&fork_snapshot_path)?;
+
+            println!("Restoring genesis file");
+            let genesis_path = home_dir.join(near_config.config.genesis_file.clone() + ".backup");
+            let original_genesis_path = home_dir.join(&near_config.config.genesis_file);
+            std::fs::rename(&genesis_path, &original_genesis_path)?;
+            return Ok(());
+        }
+        let near_config = load_config(home_dir, genesis_validation)
+            .unwrap_or_else(|e| panic!("Error loading config: {:#}", e));
+
+        if !near_config.config.store.flat_storage_creation_enabled {
+            println!(
+                "Flat storage is disabled, so the forked network will also not have flat storage"
+            );
+        }
+
+        if near_config.validator_signer.is_none() {
+            panic!("Validator ID is required; please set up validator_keys.json");
+        }
+
+        let store_opener = NodeStorage::opener(
+            home_dir,
+            near_config.config.archive,
+            &near_config.config.store,
+            None,
+        );
+
+        let store_path =
+            home_dir.join(near_config.config.store.path.clone().unwrap_or(PathBuf::from("data")));
+        let migration_snapshot_path = store_path.join(
+            near_config
+                .config
+                .store
+                .migration_snapshot
+                .get_path(&store_path)
+                .expect("Migration snapshot must be enabled"),
+        );
+        let fork_snapshot_path = store_path.join("fork-snapshot");
```

Comment:
It may make sense to mention in README that an extra disk space to store copy of state is required.